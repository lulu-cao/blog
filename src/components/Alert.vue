<template>
  <div class="w-full min-h-fit fixed bottom-0">
    <button @click="closeAlert" class="text-white mr-1 ml-auto block" >x</button>
    <p class="text-white text-center mb-3">
      <slot>Success!</slot>
    </p>
  </div>
</template>

<script>
import { onMounted } from 'vue';
export default {
  emits: ["closeAlert"],
  setup(_, { emit }) {
    const closeAlert = () => {
      emit("closeAlert");
    };

    onMounted(()=>{
      setTimeout(() => {
        emit("closeAlert");
      }, 5000);
    })

    return { closeAlert }
  }
}
</script>