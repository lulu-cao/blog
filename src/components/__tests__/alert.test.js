import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'
import Alert from '../Alert.vue'

describe('Alert component', () => {
  test('renders the correct props', () => {
    const title = 'Failure'
    const text = 'An error occurred'
    const icon = '$error'
    const type = 'error'
    const wrapper = mount(Alert, {
      props: { title, text, icon, type }
    })
    expect(wrapper.html()).toMatchSnapshot()
  },

  test('emits close event after 5 seconds', async () => {
    const wrapper = mount(Alert)
    await new Promise(resolve => setTimeout(resolve, 5000))
    expect(wrapper.emitted()).toHaveProperty("closeAlert")
  })
)})
