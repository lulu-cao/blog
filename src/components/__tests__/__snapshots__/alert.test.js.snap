// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert component > renders the correct props 1`] = `"<v-alert text="An error occurred" icon="$error" title="Failure" type="error" close-icon="$close"></v-alert>"`;

exports[`Alert component > renders the correct props for error 1`] = `"<v-alert text="An error occurred" icon="$error" title="Failure" type="error" close-icon="$close"></v-alert>"`;
